### ISSUE TITLE
PHP Warning: class_parents()

### ISSUE THREAD

**Issue by matjazpotocnik** [processwire/processwire-issues#1071]
I installed this demo module:

```
<?php namespace Aaa;

use Processwire\Process;

class Aaa extends Process {
	public static function getModuleInfo() {
		return array(
			'title' => 'Aaa',
			'summary' => 'summary',
			'version' => 1,
			);
	}
}
```

When I edit my profile OR edit a field of type FieldtypeModule (ModuleReference), like admin_theme or process OR edit a page that uses template that has this type of field, I get this:

`PHP Warning: class_parents(): Class ProcessWire\Aaa does not exist and could not be loaded in C:\inetpub\wwwroot\wire\core\Functions.php:955`

What am I doing wrong?

**Comment by teppokoivula** [comment]
It's a namespace issue. Check out this discussion for some ideas how to resolve this: https://processwire.com/talk/topic/14093-custom-namespace-in-modules/.

I've never tried the classLoader approach for module class itself, so I can only second what Mike suggested in that thread:

> I think that PW modules should remain in whatever namespace PW is using. In the case of 2.8, that is the root namespace. In the case of 3.0, that is \ProcessWire. Then, include a directory namespace (such as src), and use your namespaces there only. This seems most logical to me...

**Comment by ryancramerdesign** [comment]
@matjazpotocnik @teppokoivula Thanks, I've pushed a fix for this issue. I agree it's good to have a module in the ProcessWire namespace, since module names must already be unique. But a module should also be able to declare its own namespace when useful. That was the intention at least. It appears the primary issue was just that FieldtypeModule needed an update to better recognize when a module is using a different namespace, but I've made some other related upgrades.
